Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -105,12 +105,33 @@ ROMol *constructMolFromString(const std::string &txt,
 ChemicalReaction *RxnSmartsToChemicalReaction(
     const std::string &text, std::map<std::string, std::string> *replacements,
     bool useSmiles) {
-  std::size_t pos1 = text.find('>');
-  std::size_t pos2 = text.rfind('>');
-  if (pos1 == std::string::npos) {
+
+  std::size_t pos1;
+  std::size_t pos2;
+
+  std::vector<std::size_t> pos;
+  std::vector<std::size_t> rxnDelimiterPos;
+
+  for (std::size_t i = 0; i < text.length(); i++) {
```

Comment:
You don't need to do both loops (the one here and the one at line 126). They can both be condensed into one:
```
for (std::size_t i = 1; i < text.length(); i++) {
    if (text[i] == '>' && text[i-1]!='-') {
      pos.push_back(i);
    }
}

```